fw/tls.c

```diff
@@ -187,7 +185,8 @@ tfw_tls_connection_recv(TfwConn *conn, struct sk_buff *skb)
 			tfw_tls_purge_io_ctx(&tls->io_in);
 			kfree_skb(nskb);
 			spin_unlock(&tls->lock);
-			return T_BAD;
+			return tfw_handle_error(T_BAD, save_err_code,
```

Comment:
Why we calling `tfw_handle_error()` instead of return `SS_BLOCK_WITH_RST`? It looks confusing, we must drop connection in case of such errors.

Reply:
This function returns T_BAD if it first error (in this case we send tcp_shutdown as it was before) or BLOCK_WITH_RST if it is a seconf error